Userland/Services/RequestServer/main.cpp

```diff
@@ -40,20 +40,25 @@ ErrorOr<int> serenity_main(Main::Arguments)
     // FIXME: Establish a connection to LookupServer and then drop "unix"?
     TRY(Core::System::unveil("/tmp/portal/lookup", "rw"));
     TRY(Core::System::unveil("/etc/timezone", "r"));
+    TRY(Core::System::unveil("/tmp/session/%sid/portal/request", "rw"));
     if constexpr (TLS_SSL_KEYLOG_DEBUG)
         TRY(Core::System::unveil("/home/anon", "rwc"));
     TRY(Core::System::unveil(nullptr, nullptr));
 
-    [[maybe_unused]] auto gemini = make<RequestServer::GeminiProtocol>();
-    [[maybe_unused]] auto http = make<RequestServer::HttpProtocol>();
-    [[maybe_unused]] auto https = make<RequestServer::HttpsProtocol>();
+    [[maybe_unused]] auto gemini = make<RequestServer::GeminiProtocol>().leak_ptr();
+    [[maybe_unused]] auto http = make<RequestServer::HttpProtocol>().leak_ptr();
+    [[maybe_unused]] auto https = make<RequestServer::HttpsProtocol>().leak_ptr();
 
-    auto client = TRY(IPC::take_over_accepted_client_from_system_server<RequestServer::ConnectionFromClient>());
+    auto server = TRY(Core::LocalServer::try_create());
+    TRY(server->take_over_from_system_server());
 
-    auto result = event_loop.exec();
+    server->on_accept = [&](NonnullOwnPtr<Core::LocalSocket> client_socket) {
+        (void)IPC::new_client_connection<RequestServer::ConnectionFromClient>(move(client_socket));
+    };
 
-    // FIXME: We exit instead of returning, so that protocol destructors don't get called.
-    //        The Protocol base class should probably do proper de-registration instead of
-    //        just VERIFY_NOT_REACHED().
-    exit(result);
```

Comment:
The problem referenced by this FIXME still exists,, you're just leaking a pointer to the Protocol implementations instead of explicitly trying to avoid calling their destructors. So the FIXME should say, just moved up

Reply:
After #18985 was merged, this commit is not even necessary. Still we can argue if `RequestServer` should be spawned per logged-in user session or not. If you think we shouldn't do this, I will just drop this commit :)

Reply:
The commit looks reasonable, but keep the FIXME.